selfservice/strategy/code/strategy_recovery.go

```diff
@@ -394,20 +208,33 @@ func (s *Strategy) recoveryIssueSession(w http.ResponseWriter, r *http.Request,
 	}
 	sf.RequestURL, err = x.TakeOverReturnToParameter(f.RequestURL, sf.RequestURL, returnTo)
 	if err != nil {
-		return s.retryRecoveryFlowWithError(w, r, flow.TypeBrowser, err)
+		return s.retryRecoveryFlow(w, r, f.Type, RetryWithError(err))
 	}
 
 	config := s.deps.Config()
 
 	sf.UI.Messages.Set(text.NewRecoverySuccessful(time.Now().Add(config.SelfServiceFlowSettingsPrivilegedSessionMaxAge(ctx))))
 	if err := s.deps.SettingsFlowPersister().UpdateSettingsFlow(r.Context(), sf); err != nil {
-		return s.retryRecoveryFlowWithError(w, r, f.Type, err)
-	}
-
-	if x.IsJSONRequest(r) {
-		s.deps.Writer().WriteError(w, r, flow.NewBrowserLocationChangeRequiredError(sf.AppendTo(s.deps.Config().SelfServiceFlowSettingsUI(r.Context())).String()))
+		return s.retryRecoveryFlow(w, r, f.Type, RetryWithError(err))
+	}
+
+	if s.deps.Config().UseContinueWithTransitions(ctx) {
+		switch {
+		case f.Type.IsAPI():
+			fallthrough
+		case x.IsJSONRequest(r):
+			f.ContinueWith = append(f.ContinueWith, flow.NewContinueWithSettingsUI(sf))
```

Comment:
@aeneasr since you now need to enable the continue_with explicitely we now return the flow for both API and SPA flows to align their behavior. As @hperl request. LMKWYT.